Scripts/run_scVI.py

```diff
@@ -109,17 +111,20 @@ def run_scVI(DataPath, LabelsPath, CV_RDataPath, OutputDir, GeneOrderPath = "",
     tr_time = pd.DataFrame(tr_time)
     ts_time = pd.DataFrame(ts_time)
 
-    
-    if (NumGenes == 0):  
-        truelab.to_csv("scVI_True_Labels.csv", index = False)
-        pred.to_csv("scVI_Pred_Labels.csv", index = False)
-        tr_time.to_csv("scVI_Training_Time.csv", index = False)
-        ts_time.to_csv("scVI_Testing_Time.csv", index = False)
+    OutputDir = Path(OutputDir)
+    if not os.path.exists(OutputDir):

```

Comment:
You don't need `os.path.exists` the Path object has a function for that `OutputDir.is_file()`.
We actually don't need this if statement at all, as snakemake will make the output directory for us.